### DISCUSSION TITLE
Is there a reason why circuit::layouts::neg does not set a selector?

### DISCUSSION THREAD

**Discussion by Mickanos** [zkonduit/ezkl#196] [Q&A] [answered]
Hi, I've been reading the code of ezkl to try and learn how the protocol works in depths.
In the file circuit::layouts, the function neg (line 471 at the current commit) assigns the values -b to the output, but does not turn on the selector for the constraint corresponding to the Neg BaseOp.
Was this forgotten in the code or is there something I don't understand here?

**Comment by alexander-camuto** [comment, marked as answer]
hey ! had just rolled out that op and had forgotten to put the selector in 😬. I've added it to #194. Thanks for catching that

**Reply by alexander-camuto** [reply]
#194 significantly reduces the size of the codebase more broadly. so hopefully that should make it more legible. lmk if you have any other Qs

**Reply by Mickanos** [reply]
Sure, thank you for the quick reply.
